assembl/static2/js/app/pages/brightMirrorFiction.jsx

```diff
@@ -6,91 +6,207 @@ import { I18n } from 'react-redux-i18n';
+  /** Comment body content */
+  body: string,
+  /** Comment content locale */
+  contentLocale: string,
+  /** Comment idea identifier */
+  ideaId: string,
+  /** Comment parent identifier */
+  parentId: string
+};
+
+type LocalBrightMirrorFictionProps = BrightMirrorFictionProps & BrightMirrorFictionReduxProps & BrightMirrorFictionGraphQLProps;
+
+type BrightMirrorFictionState = {
   /** Fiction title */
   title: string,
   /** Fiction content */
   content: string,
+  /** GraphQL loading flag */
+  loading: boolean,
   /** Fiction publication state */
   publicationState: string
 };
 
-const EMPTY_STRING = '';
-
-export class BrightMirrorFiction extends Component<Props, State> {
-  constructor(props: Props) {
+export class BrightMirrorFiction extends Component<LocalBrightMirrorFictionProps, BrightMirrorFictionState> {
+  // Lifecycle functions
+  constructor(props: LocalBrightMirrorFictionProps) {
     super(props);
     this.state = {
-      title: props.data.fiction.subject || EMPTY_STRING,
-      content: props.data.fiction.body || EMPTY_STRING,
-      publicationState: props.data.fiction.publicationState || PublicationStates.PUBLISHED
+      title: EMPTY_STRING,
+      content: EMPTY_STRING,
+      loading: props.brightMirrorFictionData.loading,
+      publicationState: PublicationStates.PUBLISHED
     };
   }
 
+  componentWillReceiveProps(nextProps: LocalBrightMirrorFictionProps) {
+    // Sync state
+    this.setState({
+      title: nextProps.brightMirrorFictionData.fiction.subject ? nextProps.brightMirrorFictionData.fiction.subject : EMPTY_STRING,
+      content: nextProps.brightMirrorFictionData.fiction.body ? nextProps.brightMirrorFictionData.fiction.body : EMPTY_STRING,
+      loading: nextProps.brightMirrorFictionData.loading || nextProps.ideaWithCommentsData.loading,
+      publicationState: nextProps.brightMirrorFictionData.fiction.publicationState
+        ? nextProps.brightMirrorFictionData.fiction.publicationState
+        : PublicationStates.PUBLISHED
+    });
+  }
+
+  // Define callback functions
+  submitCommentHandler = (comment: string) => {
+    displayAlert('success', I18n.t('loading.wait'));
+
+    // Define variables
+    const { contentLocale, themeId, fictionId, createComment, ideaWithCommentsData } = this.props;
+    const createPostInputs: CreateCommentInputs = {
+      body: comment,
+      contentLocale: contentLocale,
+      ideaId: themeId,
+      parentId: fictionId
+    };
+
+    // Call the mutation function to create a comment
+    createComment({ variables: createPostInputs })
+      .then(() => {
+        // If needed post result can be fetch with `result.data.createPost.post`
+        displayAlert('success', I18n.t('debate.thread.postSuccess'));
+        // Set state here to update UI
+
+        // Refetch 'ideaWithCommentsData' to display the updated list of comments
+        ideaWithCommentsData.refetch();
+      })
+      .catch((error) => {
+        displayAlert('danger', `${error}`);
```

Comment:
I saw that we do `${error}` in many displayAlert calls but I think just `error` works fine. Is the template string adding anything? The syntax looks a bit odd

Reply:
Actually no, you need error.message or `${error}` (which will cast the object to a string, giving actually error.message), not just `error` (an object).

Reply:
Oh, I didn't know this use of template strings, good to know, thanks!